hibike/devices/ServoControl/servo_control.cpp

```diff
@@ -33,36 +31,40 @@ void loop() {
 
 // you must implement this function. It is called when the device receives a DeviceUpdate packet.
 // the return value is the value field of the DeviceRespond packet hibike will respond with
-uint32_t device_update(uint8_t param, uint32_t value) {
+uint32_t device_write(uint8_t param, uint8_t* value, size_t len) {
+  if(param< NUM_PINS && len< sizeof(servos[param]) || len< sizeof(toggle0)){
+    return 0;
+  }
+
   if (param < NUM_PINS) {
-    servos[param].write(value);
-    return servos[param].read();
+    servos[param].write((value[0]);
```

Comment:
Oh wait, the parens are mismatched here! Does this file even compile? And if it doesn't, then has the code ever been tested on hardware?